### ISSUE TITLE
Please add support for PRODUCT

### ISSUE THREAD

**Issue by kcaponeAP** [activeprospect/leadconduit-integration-leadpoint#38]
LeadPoint is currently sending  a field called product. This contains information regarding the loan type. It has no allowable values that I can spot. Recent leads are passing only the value REFINANCE.

Can we please have this added to the LeadPoint integration. Clients are unsure of what type of loan they are being sent from the vendor.

Sample lead:
https://next.leadconduit.com/events/55d492c675daefe917c20613?source_id=53d7ba989869170001000002&type=source

Please prioritize this @alexj as this is a customer request.

**Comment by kcaponeAP** [comment]
The only other plan here to move this forward is to see if LeadPoint can just send this type of information in the field 'LOAN_PURP' as the integration was built to support this. thoughts?

**Comment by zalary** [comment]
this value is currently being captured in the Mortgage Loan Amortization field.

**Comment by kcaponeAP** [comment]
Ah  long day. That means I can solve this with a mapping. Thanks.

**Comment by kcaponeAP** [comment]
Hang on, I don't want the value FIXED, which is the amortization type. I want the value for refinance. Where is that being mapped to from the product field? I do not see it.

**Comment by zalary** [comment]
oh sorry, my bad!  looking

**Comment by zalary** [comment]
you're right it's not collected, but should be and i agree it should be put in the loan purpose field.  I can work on getting this updated

**Comment by zalary** [comment]
sorry for the quickfire updates.  do you have a list of the possible values they will pass us?  the standard mortgage loan purpose field has 4 set values so we'll need to take the possible options and categorize them as such.

**Comment by kcaponeAP** [comment]
I'm on it. I have an email into them to answer that + we need a copy of their post docs. Back tomorrow with an update.

**Comment by alexj** [comment]
Thanks for jumping on this Zal, before I even noticed that Karalyn had tagged me on it.

**Comment by kcaponeAP** [comment]
Here's the information I got from them;

"The value can be customized to anything, so there are infinite exact values that can be in there.The default values will depend on what we're working with. 

For refinance data leads, it will be BP_REFI. 
For home purchase data leads, the values will be BP_NEWHOME.
For voice leads, the values use the format: BP_VOICE_REFI, BP_VOICE_NEWHOME. It can also be Refinance."

So basically, there are default values, but there can be anything in there honestly. "

**Comment by kcaponeAP** [comment]
I also asked: "why can't this be in the loan_purpose" field. here's the allowed values for loan_purpose if this helps:

Loan Purpose is something like "The Purpose of this loan is to:"
* lower_payment
* lower_rate
* change_terms
* pay_off_debt
* make_improvements
* take_out_cash

**Comment by zalary** [comment]
Is there value in trying to shoehorn the loan type data into the standard mortgage loan purpose field, or should we simply assign this to a leadpoint specific field, like leadpoint.product (similar to costco.product)

By "shoe-horning" I mean using regex so that any string that contains NEWHOME is categorized as 'purchase', anything containing REFI (including the string Refinance] would become 'refi', anything else would become 'other').

The advantage of the above means that users can make easier flow decisions based on the data (do this if it's a refi, do this if it's a new home).  The disadvantage is that we might mis-categorize or not categorize a given response as expected if it said something like New House Purchase instead of NEWHOME.

cc @alexj

**Comment by kcaponeAP** [comment]
im leaning toward leadpoint.product, but how would that tie into Velocify?

**Comment by alexkwolfe** [comment]
> The disadvantage is that we might mis-categorize or not categorize a given response as expected if it said something like New House Purchase instead of NEWHOME.

What does this mean?

**Comment by zalary** [comment]
If they can provide any string in that field and we need to map to 1 of 3 set options we can't guarantee everything ends up matched correctly, though I think we could get pretty close.

**Comment by alexkwolfe** [comment]
Do they provide a loan type in any other way? If not, then I'd imagine we should populate this field. I agree with Karalyn that we should also populate the `product` field with the raw value.

**Comment by zalary** [comment]
From the leads I've looked at it doesn't look like they pass the loan type in any other way.

So the plan is to create and populate a leadpoint.product field with the raw value 

and 

populate the mortgage.loan.type field with the best matched option available there?

**Comment by alexkwolfe** [comment]
There is some talk about having adding a generic `product` field. If we had that, maybe we would use it for this?

**Comment by kcaponeAP** [comment]
(BTW: I've asked for posting docs, but they have not sent them yet)

**Comment by kcaponeAP** [comment]
Howdy -- was this added to staging / production? I wasn't exactly sure where things were left off and need to update the client.

**Comment by alexj** [comment]
This will be added as a new field, alongside several others soon - we're aiming for this week.

**Comment by cgrayson** [comment]
This will be included in [fields issue 48](https://github.com/activeprospect/leadconduit-fields/issues/48).

**Comment by cgrayson** [comment]
Whoops, shouldn't have closed this - the field will be added by that, but this integration still needs to be updated to use that.

**Comment by zalary** [comment]
so, to clarify the next steps:

populate the newly created standard field `product` with the raw value

and

populate the mortgage.loan.type field with the best matched option available there?

**Comment by cgrayson** [comment]
> populate the mortgage.loan.type field with the best matched option available there?

Reviewing this thread, it should be `mortgage.loan.purpose`, not `.type`, right?

**Comment by zalary** [comment]
yes, `mortgage.loan.purpose`  

i confused myself somewhere along the way
